imap_processing/tests/codice/test_codice_l1a.py

```diff
@@ -2,43 +2,143 @@
 
 from pathlib import Path
 
+import cdflib
+import numpy as np
 import pytest
+import xarray as xr
 
 from imap_processing import imap_module_directory
 from imap_processing.codice.codice_l0 import decom_packets
 from imap_processing.codice.codice_l1a import process_codice_l1a
 
-TEST_DATA = [
-    (
-        Path(
-            f"{imap_module_directory}/tests/codice/data/raw_ccsds_20230822_122700Z_idle.bin"
-        ),
-        "imap_codice_l1a_hskp_20100101_v001.cdf",
-    ),
-    (
-        Path(f"{imap_module_directory}/tests/codice/data/lo_fsw_view_5_ccsds.bin"),
-        "imap_codice_l1a_lo-sw-species-counts_20240319_v001.cdf",
-    ),
-    (
-        Path(f"{imap_module_directory}/tests/codice/data/lo_fsw_view_6_ccsds.bin"),
-        "imap_codice_l1a_lo-nsw-species-counts_20240319_v001.cdf",
+EXPECTED_ARRAY_SHAPES = [
+    (99,),  # hskp
+    (1, 128),  # lo-sw-species-counts
+    (1, 112),  # lo-nsw-species-counts
+]
+EXPECTED_ARRAY_SIZES = [
+    123,  # hskp
+    16,  # lo-sw-species-counts
+    8,  # lo-nsw-species-counts
+]
+EXPECTED_FILENAMES = [
+    "imap_codice_l1a_hskp_20100101_v001.cdf",
+    "imap_codice_l1a_lo-sw-species-counts_20240319_v001.cdf",
+    "imap_codice_l1a_lo-nsw-species-counts_20240319_v001.cdf",
+]
+TEST_PACKETS = [
+    Path(
+        f"{imap_module_directory}/tests/codice/data/raw_ccsds_20230822_122700Z_idle.bin"
     ),
+    Path(f"{imap_module_directory}/tests/codice/data/lo_fsw_view_5_ccsds.bin"),
+    Path(f"{imap_module_directory}/tests/codice/data/lo_fsw_view_6_ccsds.bin"),
+]
+
+# Placeholder for validation data files
+VALIDATION_DATA = [
+    f"{imap_module_directory}/tests/codice/data/validation_hskp.cdf",
+    f"{imap_module_directory}/tests/codice/data/validation_lo-sw-species-counts.cdf",
+    f"{imap_module_directory}/tests/codice/data/validation_lo-nsw-species-counts.cdf",
 ]
 
 
-@pytest.mark.parametrize(("test_file", "expected_filename"), TEST_DATA)
-def test_codice_l1a(test_file: Path, expected_filename: str):
-    """Tests the ``process_codice_l1a`` function and ensure that a proper CDF
-    files are created.
+@pytest.fixture(params=TEST_PACKETS)
+def test_l1a_data(request) -> tuple[xr.Dataset, str]:
+    packets = decom_packets(request.param)
+    dataset, cdf_filename = process_codice_l1a(packets)
+    return dataset, cdf_filename
+
+
+@pytest.mark.parametrize(
+    "test_l1a_data, expected_filename",
+    list(zip(TEST_PACKETS, EXPECTED_FILENAMES)),
+    indirect=["test_l1a_data"],
+)
```

Comment:
nice refactoring and I learned something new about `indirect`.

Reply:
I agree this is a nice way of doing it! Out of curiosity, if you were to transpose the order of the variables would you need the indirect still? Going from `EXPECTED_ARRAY_SHAPES = ..., EXPECTED_ARRAY_SIZES = ...` to `HSK_EXPECTATIONS = (shape, size, ...)` and parameterize over `HSK, SCI, ...`

Reply:
Interesting idea. I can explore this in a future PR, especially when I go to add more data products.